pkg/block/index.go

```diff
@@ -239,6 +243,10 @@ func ReadIndexCache(logger log.Logger, fn string) (
 	return v.Version, symbols, lvals, postings, nil
 }
 
+func IsUnmarshalError(err error) bool {
+	return strings.Contains(err.Error(), UnmarshalErrorMsg)
```

Comment:
I think we can do better than just parsing the message.

I would propose sentinel error instead:

`IndexCacheUnmarshalError = errors.New("....")`

and then on line 201:  `errors.Wrap(IndexCacheUnmarshalError, err.Error())`

and then instead of this function user can do `if !os.IsNotExist(errors.Cause(err)) && errors.Cause(err) =! block.IndexCacheUnmarshalError{` hm? (:

The problem with string parsing is that it could be unsafe e.g we could catch some unexpected errors there. Let's be explicit.

Reply:
That sounds reasonable, I will improve it like that. (: